docs/REQUIRES_VS_IMPORTS.md

```diff
@@ -0,0 +1,470 @@
+  ],
+  
+  methods: [
+    function loadUser() {
+      // userService might be running on a different server
+      return this.userService.getCurrentUser();
+    }
+  ]
+});
+```
+
+## Best Practices
+
+### Use `requires` when:
+- Creating instances of FOAM classes
+- Building UI components that compose other views
+- Working with model classes that you need to instantiate
+- You need factory methods for object creation
+- Dependencies are known at development time
+- You want to avoid typing full package names
+
+### Use `imports` when:
+- Accessing application context or global state
+- Injecting services like logging, authentication, DAOs
+- Getting configuration values or runtime data
+- You want loose coupling and testability
+- Dependencies come from the execution environment
+- You need services that might be remote
+
+### Use `exports` when:
+- Creating services that child components need
+- Building reusable components that provide context
+- Implementing service layers
+- Creating testable architectures with dependency injection
+
+### General Guidelines:
+
+1. **Prefer `imports`** for services and runtime data - it creates more flexible, testable code
+2. **Use `requires`** for class instantiation and when you need to create multiple instances
+3. **Mark optional dependencies** with `?` in imports
+4. **Use aliasing** to avoid naming conflicts
+5. **Keep dependencies minimal** - only import what you need
+6. **Design for testability** - prefer injection over direct lookups
+7. **Think about the context chain** - exports flow down to imports
+
+## Common Patterns
+
+### Service Layer Pattern
+```javascript
+foam.CLASS({
+  name: 'UserService',
+  
+  imports: [
+    'userDAO',      // DAO injected from context
+    'log'          // Logging service
+  ],
+  
+  exports: [
+    'this as userService'  // Export self for other components
+  ],
+  
+  methods: [
+    function getCurrentUser() {
+      return this.userDAO.find(this.currentUserId);
+    }
+  ]
+});
+```
+
+### Controller Pattern
+```javascript
+foam.CLASS({
+  name: 'UserController',
+  
+  imports: [
+    'userService',   // Service injected (loose coupling)
+    'notify'        // Notification service
+  ],
+  
+  requires: [
+    'foam.u2.DetailView',  // View class for creating forms
+    'UserModel'           // Model class for data validation
+  ],
+  
+  methods: [
+    function createUserForm() {
+      // Use required class to create view
+      return this.DetailView.create({
+        data: this.UserModel.create()  // Use required model
+      });
+    },
+    
+    function saveUser(user) {
+      // Use injected service
+      return this.userService.save(user).then(() => {
+        this.notify.success('User saved!');
+      });
+    }
+  ]
+});
+```
+
+### View Pattern
+```javascript
+foam.CLASS({
+  name: 'UserView',
+  extends: 'foam.u2.View',
+  
+  imports: [
+    'data',          // Data from parent context
```

Comment:
You wouldn't import 'data', since it is a property inherited from 'foam.u2.View'